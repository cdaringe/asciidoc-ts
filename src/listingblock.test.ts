import { describe, expect, test } from "vitest";
import { toAST } from "./mod.js";

test("ListingBlock", async ({ expect }) => {
  const input = `
[foo, bar]
baz

`;
  const result = toAST(input);
  expect(result).toMatchSnapshot();
});

test("ListingBlock - delimited", async ({ expect }) => {
  const input = `
[foo, bar]
----
newlines are


ok
----
`;
  const result = toAST(input);
  expect(result).toMatchSnapshot();
});

test("CodeBlock", async ({ expect }) => {
  const input = `
[source, js]
----
foo();
----
`;
  const result = toAST(input);
  expect(result).toMatchSnapshot();
});
