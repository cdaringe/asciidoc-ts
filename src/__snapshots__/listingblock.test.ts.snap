// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CodeBlock 1`] = `
{
  "blocks": [
    {
      "content": "foo();

bar();",
      "delimited": true,
      "metadata": {
        "attributes": [
          {
            "name": "source",
            "type": "AttributeEntry",
          },
          {
            "name": "js",
            "type": "AttributeEntry",
          },
        ],
      },
      "type": "CodeBlock",
    },
  ],
  "type": "Document",
}
`;

exports[`ListingBlock - delimited 1`] = `
{
  "blocks": [
    {
      "type": "BlankLine",
    },
    {
      "content": "newlines are


ok",
      "delimited": true,
      "metadata": {
        "attributes": [
          {
            "name": "foo",
            "type": "AttributeEntry",
          },
          {
            "name": "bar",
            "type": "AttributeEntry",
          },
        ],
      },
      "type": "ListingBlock",
    },
  ],
  "type": "Document",
}
`;

exports[`ListingBlock 1`] = `
{
  "blocks": [
    {
      "type": "BlankLine",
    },
    {
      "content": [
        {
          "content": "baz",
          "type": "PlainText",
        },
      ],
      "metadata": {
        "attributes": [
          {
            "name": "foo",
            "type": "AttributeEntry",
          },
          {
            "name": "bar",
            "type": "AttributeEntry",
          },
        ],
      },
      "type": "Paragraph",
    },
  ],
  "type": "Document",
}
`;
